src/libstore/build.cc

```diff
@@ -1630,17 +1630,47 @@ void DerivationGoal::buildDone()
         registerOutputs();
 
         if (settings.postBuildHook != "") {
-            printMsg(lvlChatty, format("executing post-build hook '%1%'")
-                % settings.postBuildHook);
-            Activity act(*logger, actPostBuildHook, Logger::Fields{drvPath});
+            /* needs exception handling */
+            Activity act(*logger, lvlInfo, actPostBuildHook,
+                fmt("running post-build-hook '%s'", settings.postBuildHook),
+                Logger::Fields{drvPath});
             PushActivity pact(act.id);
             auto outputPaths = drv->outputPaths();
             std::map<std::string, std::string> hookEnvironment = getEnv();
+
             hookEnvironment.emplace("DRV_PATH", drvPath);
-            hookEnvironment.emplace("OUT_PATHS", chomp(concatStringsSep(":", outputPaths)));
+            hookEnvironment.emplace("OUT_PATHS", chomp(concatStringsSep(" ", outputPaths)));
 
             RunOptions opts(settings.postBuildHook, {});
             opts.environment = hookEnvironment;
+
+            struct LogSink : Sink {
+                Activity & act;
+                std::string currentLine;
+
+                LogSink(Activity & act) : act(act) { }
+
+                void operator() (const unsigned char * data, size_t len) override {
+                    for (size_t i = 0; i < len; i++) {
+                        auto c = data[i];
+
+                        if (c == '\n') {
+                            if (settings.verboseBuild) {
+                                printError("post-build-hook: " + currentLine);
+                            } else {
+                                act.result(resPostBuildLogLine, currentLine);
+                            }
+                            currentLine.clear();
+                        } else {
+                            currentLine += c;
+                        }
+                    }
+                }
+            };
+            LogSink sink(act);
+
+            opts.standardOut = &sink;
+            opts.standardErr = &sink;
```

Comment:
This bit doesn't seem to work. Stderr gets swallowed no matter what.

Reply:
It's because it's not being drained like stdout:
```
    if (options.standardOut)
        drainFD(out.readSide.get(), *options.standardOut);
```
However draining them both would require a `select()` loop or multiple threads. It's probably simpler to drop `options.standardErr` and replace it with a boolean flag denoting that stdout should be duped to stderr.

Reply:
ah! Yes that makes sense.